src/g_game.c

```diff
@@ -4060,7 +4060,8 @@ void doomprintf(player_t *player, msg_category_t category, const char *s, ...)
   va_list v;
 
   if ((category == MESSAGES_TOGGLE && !show_toggle_messages) ||
-      (category == MESSAGES_PICKUP && !show_pickup_messages))
+      (category == MESSAGES_PICKUP && !show_pickup_messages) ||
+      (category == MESSAGES_OBITUARY && !show_obituary_messages))
```

Comment:
If you make the categories powers of two and construct a composite `shown_categories` you could do something like `if (!(category & shown_categories))` to avoid needing to edit this list in the future.

Reply:
Right, but the `show_*_messages` keys are currently boolean variables (we decided against combining all "show this or that type of message" into one variable), so I'll have to remember to bit-shift them before compositing them - so there's still that one place in the code that I'll have to change whenever a new category of messages is added (which I am not expecteing at all anytime soon).